packages/cli/src/hooks/init/termination-signals.ts

```diff
@@ -1,20 +1,29 @@
+import {flush} from '../analytics/analytics';
 import {buildEvent} from '../analytics/eventUtils';
 
 export const exitSignals = ['SIGINT', 'SIGTERM', 'SIGQUIT', 'SIGHUP'] as const;
 
-export const handleTerminationSignals = () => {
+export function handleTerminationSignals() {
   exitSignals.forEach((signal) =>
-    process.on(signal, (_sig: string, code: number) => {
-      trackEvent(signal, code);
+    process.on(signal, async (_sig: string, exitCode: number) => {
+      await trackEvent(signal);
+      await flush();
+      /**
+       * Signal Exits: If Node.js receives a fatal signal such as SIGTERM or SIGHUP, then its exit code will be 128 plus the value of the signal code.
+       * This is a standard POSIX practice, since exit codes are defined to be 7-bit integers, and signal exits set the high-order bit,
+       * and then contain the value of the signal code.
+       *
+       * For example, signal SIGABRT has value 6, so the expected exit code will be 128 + 6, or 134.
+       * For more info, visit https://nodejs.org/api/process.html#exit-codes
+       */
+      // eslint-disable-next-line no-process-exit
+      process.exit(128 + exitCode);
```

Comment:
We talked a bit about that, I'm wondering, does oclif has its own handler that exit the process?
In which case we should do `process.exitCode = 128+exitCode` and let it do its thing

(IMO, try the snippet ☝️ . If it works, keep the snippet, otherwise, keep the process.exit())

Reply:
I tried 2 approaches

`process.exitCode = 128 + exitCode;`
Does nothing since nothing is asking the process to exit.

`cli.exit(128 + exitCode);`
Built-in method in [cli-ux](https://www.npmjs.com/package/cli-ux) to exit the process. However, it also throws an [ExitError](https://github.com/oclif/errors/blob/95b0ea57bb23d0c4fcb0d877260418387e940492/src/errors/exit.ts). But nothing is intercepting it... so we don't want that.

Oclif recommends throwing an error rather than exiting the process. However, in this case, I think it's too late to throw an error.

Reply:
👍 all good, process.exit is then the way to go.